parcel/client.py

```
@@ -98,6 +101,18 @@ def stop_timer(self, file_size=None):
             log.info(
                 'Download complete: {0:.2f} Gbps average'.format(rate))
 
+    def validate_file_md5sum(self, stream):
+        if not stream.check_file_md5sum:
+            log.debug('checksum validation disabled')
+            return
+        if not stream.is_regular_file:
+            log.debug('Not a regular file, skipping checksum validation')
```

This seems like it should be more than a debug (and by default silent) error, perhaps a fatal error?

The same argument could be made for `SegmentProducer.check_file_exists_and_size`, though I believe silently failing to check the md5sum is a more critical issue.